spec/index.html

```diff
@@ -197,22 +192,103 @@
         <pre class="idl">
         enum AutocompleteType { "none", "inline", "list", "both" };
         enum CheckedType { "false", "mixed", "true" };
+        enum CurrentType { "page", "step", "location", "date", "time", "true", "false" };
         enum InvalidType { "false", "true", "grammar", "spelling" };
         enum LiveType { "off", "polite", "assertive" };
         enum LiveRelevantType { "additions", "additions text", "all", "removals", "text" };
         enum OrientationType { "horizontal", "vertical" };
+        enum RoleType {
+          "alert",
+          "alertdialog",
+          "application",
+          "article",
+          "banner",
+          "button",
+          "cell",
+          "checkbox",
+          "columnheader",
+          "combobox",
+          "complementary",
+          "contentinfo",
+          "definition",
+          "dialog",
+          "directory",
+          "document",
+          "feed",
+          "figure",
+          "form",
+          "grid",
+          "gridcell",
+          "group",
+          "heading",
+          "img",
+          "link",
+          "list",
+          "listbox",
+          "listitem",
+          "log",
+          "main",
+          "marquee",
+          "math",
+          "menu",
+          "menubar",
+          "menuitem",
+          "menuitemcheckbox",
+          "menuitemradio",
+          "navigation",
+          "none",
+          "note",
+          "option",
+          "presentation",
+          "progressbar",
+          "radio",
+          "radiogroup",
+          "region",
+          "row",
+          "rowgroup",
+          "rowheader",
+          "scrollbar",
+          "search",
+          "searchbox",
+          "separator",
+          "slider",
+          "spinbutton",
+          "status",
+          "switch",
+          "tab",
+          "table",
+          "tablist",
+          "tabpanel",
+          "term",
+          "textbox",
+          "timer",
+          "toolbar",
+          "tooltip",
+          "tree",
+          "treegrid",
+          "treeitem"
+        };
         enum SortType { "none", "ascending", "descending", "other" };
 
         interface AccessibleNode {
-          attribute DOMString role;
+          attribute RoleType role;
+          attribute DOMString? roleDescription;
 
           // Accessible label and description.
           attribute DOMString? label;
+          attribute iterable&lt;AccessibleNode&gt; labeledBy;
```

Comment:
I didn't like making this the one glaring exception in otherwise very
straightforward translations from the ARIA attribute names. Any time we
deviate it's just adding confusion for users.

I guess labeledBy because it's different from ARIA but close enough that
it will confuse people.

I guess my preference would be to try to come up with a consistent way to
modify all of the relationship attributes, then, rather than making
labeledBy an exception.

Since it's a reference to an AccessibleNode and not a DOM element, maybe labelNode instead of labelElement?

We could add Node or Nodes to all of them to help convey whether they take a single reference or an array.

labelNode
descriptionNode
activeDescendantNode
controlledNodes
detailsNode
errorMessageNode
flowToNode
ownedNodes